sbi/simulators/user-input-checks.md

```diff
@@ -0,0 +1,62 @@
+# User input checks
+
+## What we require from the user
+
+- **simulator Callablle** that takes and returns torch tensor or np arrays
+- **prior object** with `.sample()` and `.log_prob()` methods, `batch_shape==1` (but it doesn't have to have attributes `batch_shape` and `event_shape`.  
+  - `.sample()` returns a single parameter set, `.sample((batch_size,))` returns `batch_size` parameter sets. **Note**: the sample size here will turn into the `batch_size` of the simulator, hence our use of the terminology here.
```

Comment:
I'm a bit confused here. By "`batch_size` of the simulator" you mean the `simulation_batch_size` of the `simulate_in_batches` function, no? But, as I understand the code, `simulation_batch_size` is just another variable that has nothing to do with the prior. I think we always draw **all** parameters from the prior, but then we just simulate them in batches of size `simulate_in_batches`. If this is correct, maybe just call it `num_samples` here?

Or do you mean another `batch_size`?

Reply:
Thanks Michael, this is connected with my previous comment on this topic @janfb, thus clarification is indeed needed (2 people confused).

Reply:
tldr; batch_size=simulation_batch_size. 

I think the **Note** was added by @meteore, actually. And what he means here, I think, is related to the terminology of batch (non iid), event and sample (iid) shape in PyTorch. We agreed that in our use-cases the prior should always have batch shape 1 because samples from the prior are by definition iid. However, in this PyTorch terminology the simulator will have a batch shape depending on the numbers of parameter sets passed to it because every parameter set defines a new random variable we can sample from so that simulations within a batch of parameters are always non-idd. 

For example, say we have sampled 10 parameters from a 3D Gaussian prior (batch_shape=1, sample_shape=10, even_shape=3), then we can pass these 10 samples to the simulator in one go, resulting in batch_shape=10=(simulation_batch_size), sample_shape=1; or we could pass all 10 parameters and simulate twice for every parameter resulting in batch_shape=10, sampe_shape=2. 

Do you agree?